### PR TITLE
Fixing restore and recovery bugs https://microsoft.visualstudio.com/w…

### PR DESCRIPTION
**Pull request by mkherani**
…eb/wi.aspx?pcguid=cb55739e-4afe-46a3-970f-1b49d8ee7564&id=8401714

### CODE REVIEW

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/RecoveryPointConversions.cs [hunk 1]
```diff
@@ -61,7 +64,7 @@ public class RecoveryPointConversions
                         RecoveryPointId = rp.Name,
                         BackupManagementType = item.BackupManagementType,
                         ItemName = protectedItemName,
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/RecoveryPointConversions.cs]: It should be ProtectedItemUri

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/RestoreDiskAPIs.cs [hunk 2]
```diff
@@ -63,17 +68,17 @@ public partial class ServiceClientAdapter
...
-                restoreRequest.EncryptionDetails = new EncryptionDetails()
-                {
-                    EncryptionEnabled = rp.EncryptionEnabled,
-                    KekUrl = rp.KeyAndSecretDetails.KeyUrl,
-                    KekVaultId = rp.KeyAndSecretDetails.KeyVaultId,
-                    SecretKeyUrl = rp.KeyAndSecretDetails.SecretUrl,
-                    SecretKeyVaultId = rp.KeyAndSecretDetails.SecretVaultId,
-                };              
-            }
+            //if (rp.EncryptionEnabled)
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/RestoreDiskAPIs.cs]: Delete this commented code

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/RestoreDiskAPIs.cs [hunk 3]
```diff
@@ -13,9 +13,11 @@
 // ----------------------------------------------------------------------------------
 
 using Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Helpers;
 using Microsoft.Azure.Commands.RecoveryServices.Backup.Properties;
 using Microsoft.Azure.Management.RecoveryServices.Backup.Models;
 using System;
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/RestoreDiskAPIs.cs]: Move up System usings

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/RecoveryPointConversions.cs [hunk 4]
```diff
@@ -61,7 +64,7 @@ public class RecoveryPointConversions
                         RecoveryPointId = rp.Name,
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/RecoveryPointConversions.cs]: See if we can put common part of conversion between List and Get into another function

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/RestoreDiskAPIs.cs (hunk 2 above)
Comment (mkherani) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/RestoreDiskAPIs.cs]: Done

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/RestoreDiskAPIs.cs (hunk 3 above)
Comment (mkherani) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/RestoreDiskAPIs.cs]: Its sorted by VS in alphabetical order. Not changing it.

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/RecoveryPointConversions.cs (hunk 4 above)
Comment (mkherani) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/RecoveryPointConversions.cs]: there are few additional field in detail thats why we kept it separate. I will check the possibility and do it in next checkin

**Pull request merged**